indico/modules/events/registration/util.py

```diff
@@ -58,12 +58,12 @@ def get_event_section_data(regform, management=False):
     return [s.view_data for s in regform.sections if not s.is_deleted and (management or not s.is_manager_only)]
 
 
-def make_registration_form(regform, management=False):
+def make_registration_form(regform, management=False, edit=False):
     """Creates a WTForm based on registration form fields"""
 
     class RegistrationFormWTF(IndicoForm):
         def validate_email(self, field):
-            if regform.get_registration(email=field.data):
+            if not edit and regform.get_registration(email=field.data):
```

Comment:
You could just pass the registration to `make_registration_form` (could even be instead of the `edit` arg).